crates/client/data-availability/src/avail/mod.rs

```diff
@@ -50,41 +89,47 @@ impl DaClient for AvailClient {
 }
 
 impl AvailClient {
-    async fn publish_data(&self, bytes: &BoundedVec<u8>) -> Result<H256> {
+    async fn publish_data(&self, bytes: &BoundedVec<u8>) -> Result<()> {
+        let mut ws_client = self.ws_client.lock().await;
+
         let data_transfer = AvailApi::tx().data_availability().submit_data(bytes.clone());
         let extrinsic_params = AvailExtrinsicParams::new_with_app_id(self.app_id);
-        let events = self
-            .ws_client
-            .tx()
-            .sign_and_submit_then_watch(&data_transfer, &self.signer, extrinsic_params)
-            .await?
-            .wait_for_finalized_success()
-            .await?;
-
-        Ok(events.block_hash())
-    }
 
-    async fn verify_bytes_inclusion(&self, block_hash: H256, bytes: &BoundedVec<u8>) -> Result<()> {
-        let submitted_block = self
-            .ws_client
-            .rpc()
-            .block(Some(block_hash))
-            .await?
-            .ok_or(anyhow::anyhow!("Invalid hash, block not found"))?;
-
-        submitted_block
-            .block
-            .extrinsics
-            .into_iter()
-            .filter_map(|chain_block_ext| AppUncheckedExtrinsic::try_from(chain_block_ext).map(|ext| ext.function).ok())
-            .find(|call| match call {
-                Call::DataAvailability(DaCall::submit_data { data }) => data == bytes,
-                _ => false,
-            })
-            .ok_or(anyhow::anyhow!("Bytes not found in specified block"))?;
+        match ws_client.client().tx().sign_and_submit(&data_transfer, &self.signer, extrinsic_params).await {
+            Ok(i) => i,
+            Err(e) => {
+                if e.to_string().contains("restart required") {
+                    ws_client.restart().await;
```

Comment:
```suggestion
                    ws_client.restart().await?;
```

Reply:
I left it this way, so the original error is propagated, and not the restart failing error. What do you think?

Reply:
Let's add a comment then until we move away from anyhow

Reply:
I kept it this way, so the original error is propagated back to the worker.